### DISCUSSION TITLE
Consequence of enforceActions: 'never'

### DISCUSSION THREAD

**Discussion by mekanix** [mobxjs/mobx#2932] [General] [answered]
I couldn't find what changes with this config option. I mean, I know why "always" is great: makes your team change store from predefined place, but what is the negative side of "never"? Thank you!

**Comment by urugator** [comment, marked as answer]
`action` provides batching. By using `never` it's easier to forget `action` and therefore risking suboptimal perf.

**Reply by mekanix** [reply]
Can you suggest where I could read more about it? If source, then maybe I could document it for others with the same question? I would like to understand mobx as much as I can, and actions are what I'm currently researching. Thank you!

**Reply by urugator** [reply]
https://mobx.js.org/actions.html

> They are run inside transactions. No observers will be updated until the outer-most action has finished, guaranteeing that  intermediate or incomplete values produced during an action are not visible to the rest of the application until the action has completed.

**Reply by mekanix** [reply]
Sorry, my bad, I didn't understand that section as I should. Thank you for clarification!
